requires.py

```diff
@@ -1,113 +1,288 @@
-        return client_cert, client_key
+        """
+        Deprecated.  Use [request_client_cert][] and the [client_certs][]
+        collection instead.
+
+        Return a globally shared client certificate and key.
+        """
+        data = self.all_joined_units.received_raw
+        return (data['client.cert'], data['client.key'])
 
     def get_server_cert(self):
-        '''Return the server certificate and key from the relation objects.'''
-        conversation = self.conversation()
-        # Get the name of the unit this code is running on.
+        """
+        Deprecated.  Use the [server_certs][] collection instead.
+
+        Return the cert and key of the first server certificate requested.
+        """
+        if not self.server_certs:
+            return (None, None)
+        cert = self.server_certs[0]
+        return (cert.cert, cert.key)
+
+    @property
+    def server_certs(self):
+        """
+        List of [Certificate][] instances for all available server certs.
+        """
+        certs = []
         name = hookenv.local_unit().replace('/', '_')
-        # Prefix the keys with name so each unit can get unique certs and keys.
-        server_cert = conversation.get_remote('{0}.server.cert'.format(name))
-        server_key = conversation.get_remote('{0}.server.key'.format(name))
-        return server_cert, server_key
-
-    def request_server_cert(self, cn, sans, cert_name):
-        '''Set the CN, list of sans, and certifiicate name on the relation to
-        request a server certificate.'''
-        conversation = self.conversation()
-        # A server certificate requires a CN, sans, and a certificate name.
-        conversation.set_remote('common_name', cn)
-        conversation.set_remote('sans', json.dumps(sans))
-        conversation.set_remote('certificate_name', cert_name)
+        raw_data = self.all_joined_units.received_raw
+        json_data = self.all_joined_units.received
 
-    def add_request_server_cert(self, cn, sans):
-        conversation = self.conversation()
-        cert_requests = conversation.get_local('cert_requests')
+        # for backwards compatibility, the first cert goes in its own fields
+        common_name = self.relations[0].to_publish_raw['common_name']
+        cert = raw_data['{}.server.cert'.format(name)]
+        key = raw_data['{}.server.key'.format(name)]
+        if cert and key:
+            certs.append(Certificate('server',
+                                     common_name,
+                                     cert,
+                                     key))
 
-        if cert_requests:
-            cert_requests[cn] = {'sans': sans}
-        else:
-            cert_requests = {
-                cn: {'sans': sans}}
-        conversation.set_local('cert_requests', cert_requests)
+        # subsequent requests go in the collection
+        certs_data = json_data['{}.processed_requests'.format(name)] or {}
```

Comment:
I place the earlier "Clever" remark on hold until you riddle me this: where does `json_data` omit the first cert from `raw_data`?

Reply:
+1, i didn't grok `request_server_cert` when i made the comment in `server_certs`.